src/tests.rs

```diff
@@ -83,3 +83,372 @@ fn number_of_children(n_threads: usize) {
+        assert!(child.generate_node_ids);
+    }
+}
+
+#[test]
+fn test_graph_ids_disabled() {
+    let game = TenMoveGame::default();
+    let mut ismcts = IsmctsHandler::new(game);
+    ismcts.run_iterations(1, 100);
+
+    // Check that root node has an empty ID
+    assert!(ismcts.root_node.id.is_none());
+    assert!(!ismcts.root_node.generate_node_ids);
+
+    // Check that children have empty IDs
+    let children = ismcts.root_node.children.read().unwrap();
+    for child in children.iter() {
+        assert!(child.id.is_none());
+        assert!(!child.generate_node_ids);
+    }
+}
+
+#[test]
+fn test_policy_priors_are_stored() {
+    let game = GameWithPolicy::default();
+    let mut ismcts = IsmctsHandler::new(game);
+    ismcts.run_iterations(1, 100);
+
+    let children = ismcts.root_node.children.read().unwrap();
+
+    // Should have explored all 3 moves
+    assert_eq!(3, children.len());
+
+    // Find each move and check its prior probability
+    for child in children.iter() {
+        let stats = child.statistics.read().unwrap();
+        let mov = child.mov.as_ref().unwrap();
+
+        match mov {
+            0 => assert!((stats.prior_probability - 0.7).abs() < 0.001),
+            1 | 2 => assert!((stats.prior_probability - 0.15).abs() < 0.001),
+            _ => panic!("Unexpected move"),
+        }
+    }
+}
+
+#[test]
+fn test_policy_affects_selection() {
+    let game = GameWithPolicy::default();
+    let mut ismcts = IsmctsHandler::new(game);
+    ismcts.run_iterations(1, 1000);
+
+    let children = ismcts.root_node.children.read().unwrap();
+
+    // Find visit counts for each move
+    let mut visits = std::collections::HashMap::new();
+    for child in children.iter() {
+        let mov = *child.mov.as_ref().unwrap();
+        let visit_count = child.statistics.read().unwrap().visit_count;
+        visits.insert(mov, visit_count);
+    }
+
+    // Move 0 should have significantly more visits than moves 1 and 2
+    // due to its higher prior (0.7 vs 0.15)
+    assert!(visits[&0] > visits[&1]);
+    assert!(visits[&0] > visits[&2]);
+}
+
+#[test]
+fn test_value_network_is_used() {
+    let game = GameWithValue::default();
+    let mut ismcts = IsmctsHandler::new(game);
+    ismcts.run_iterations(1, 100);
+
+    let children = ismcts.root_node.children.read().unwrap();
+
+    // All children should have been explored
+    assert!(children.len() > 0);
+
+    // Check that value estimates are being used by verifying
+    // that rewards are accumulating based on the value estimate (0.5)
+    for child in children.iter() {
+        let stats = child.statistics.read().unwrap();
+        if stats.visit_count > 0 {
+            let avg_reward = stats.reward / stats.visit_count as f64;
+            // The average reward should be close to the value estimate
+            // (allowing for some variance due to the stochastic nature)
+            assert!((0.0..=1.0).contains(&avg_reward));
```

Comment:
Isn't `0.0..=1.0` the entire possible range? i.e. not necessarily close to the estimate